src/App.tsx

```diff
@@ -1,26 +1,304 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import { addTodo, deleteTodo, editTodo, getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-const USER_ID = 0;
+interface Errors {
+  messages: string[];
+}
 
 export const App: React.FC = () => {
+  const [isClearCompletedDesabled, setIsClearCompletedDesabled] =
+    useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errors, setErrors] = useState<Errors>({ messages: [] });
+  const [input, setInput] = useState('');
+  const [isLoadingTodos, setIsLoadingTodos] = useState(true);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
```

Comment:
```suggestion
  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
```
Probably they are ids as it's an array of numbers